scipy/optimize/cobyla.py

```diff
@@ -231,12 +232,29 @@ def _minimize_cobyla(fun, x0, args=(), constraints=(),
         if 'args' not in con:
             con['args'] = ()
 
-    m = len(constraints)
+    # m is the total number of constraint values
+    # it takes into account that some constraints may be vector-valued
+    m = 0
+    for c in constraints:
```

Comment:
I'd suggest to avoid some code duplication by calculating length of each constraint here,
```
cons_lengths = []
for c in constraints:
    ...
    cons_lengths.append(...)
m = sum(cons_lengths)
```
and in calcfc
```
i = 0
for size, c in izip(cons_length, constraints):
    con[i:i+size] = c['fun'](x, *c['args'])
    i += size
```
and add to the imports on top of file
```
try:
    from itertools import izip
except ImportError:
    izip = zip
```
The code inside calcfc is performance-critical so it may make sense to do some micro-optimization...